tst/com/amazon/corretto/crypto/provider/test/AesKeyWrapTest.java

```diff
@@ -0,0 +1,550 @@
+    }
+    int finalOutputSize = c.getOutputSize(0);
+    byte[] ciphertext = c.doFinal();
+    assertEquals(ciphertext.length, finalOutputSize);
+
+    c.init(Cipher.DECRYPT_MODE, kek);
+    // first pass, no buffered data
+    for (int inputSize : inputSizes) {
+      assertTrue(c.getOutputSize(inputSize) == Math.max(inputSize - 8, 8));
+    }
+    // second pass, buffer data
+    for (int inputSize : inputSizes) {
+      c.update(new byte[inputSize]);
+    }
+    // reset and update w/ ciphertext otherwise decrypt will fail.
+    c.init(Cipher.DECRYPT_MODE, kek);
+    c.update(ciphertext);
+    finalOutputSize = c.getOutputSize(0);
+    assertTrue(c.doFinal().length <= finalOutputSize);
+  }
+
+  @Test
+  public void testEngineGetParametersAndIv() throws Exception {
+    final IvParameterSpec iv = TestUtil.genIv(1, 16);
+    final Cipher c = getCipher(TestUtil.NATIVE_PROVIDER);
+    assertTrue(c.getParameters() == null);
+    assertTrue(c.getIV() == null);
+    final int[] modes =
+        new int[] {
+          Cipher.ENCRYPT_MODE, Cipher.ENCRYPT_MODE, Cipher.ENCRYPT_MODE, Cipher.ENCRYPT_MODE
+        };
+    for (int mode : modes) {
+      c.init(mode, getAesKey(128 / 8));
+      assertTrue(c.getParameters() == null);
+      assertTrue(c.getIV() == null);
+      c.init(mode, getAesKey(128 / 8), (AlgorithmParameterSpec) null, (SecureRandom) null);
+      assertTrue(c.getParameters() == null);
+      assertTrue(c.getIV() == null);
+      c.init(mode, getAesKey(128 / 8), (AlgorithmParameters) null, (SecureRandom) null);
+      assertTrue(c.getParameters() == null);
+      assertTrue(c.getIV() == null);
+      c.init(mode, getAesKey(128 / 8), iv);
+      assertTrue(c.getParameters() != null);
+      assertTrue(c.getIV() != null);
+      assertArraysHexEquals(c.getIV(), iv.getIV());
+    }
+  }
+
+  @Test
+  public void testUpdateWithOversizedInputLen() throws Exception {
+    final Cipher c = getCipher(TestUtil.NATIVE_PROVIDER);
+    final Key kek = getAesKey(128 / 8);
+    c.init(Cipher.ENCRYPT_MODE, kek);
+    final byte[] plaintext = TestUtil.getRandomBytes(1024);
+
+    assertThrows(
+        IllegalArgumentException.class, () -> c.update(plaintext, 0, plaintext.length + 8));
+    assertThrows(IllegalArgumentException.class, () -> c.update(plaintext, 1, plaintext.length));
+    assertThrows(
+        IllegalArgumentException.class, () -> c.update(plaintext, plaintext.length + 8, 0));
+
+    c.update(plaintext, 0, plaintext.length);
+    final byte[] ciphertext = c.doFinal();
+
+    c.init(Cipher.DECRYPT_MODE, kek);
+    assertThrows(
+        IllegalArgumentException.class, () -> c.doFinal(ciphertext, 0, ciphertext.length + 8));
+    assertThrows(IllegalArgumentException.class, () -> c.doFinal(ciphertext, 1, ciphertext.length));
+    assertThrows(
+        IllegalArgumentException.class, () -> c.doFinal(ciphertext, ciphertext.length + 8, 0));
+    assertArraysHexEquals(plaintext, c.doFinal(ciphertext, 0, ciphertext.length));
+  }
```

Comment:
This test does not appear to test any unique logic inside `AesKeyWrapSpi`. Suggest removing it.